pulsar-client-cpp/include/pulsar/MessageId.h

```diff
@@ -34,6 +34,16 @@ class MessageId {
     MessageId& operator=(const MessageId&);
     MessageId();
 
+    /**
+     * Only for MultiTopicsConsumer to set a valid topicName
+     */
+    void setTopicName(std::string topicName);
+
+    /**
+     * Only for MultiTopicsConsumer to get a valid topicName
+     */
+    std::string& getTopicName() const;
```

Comment:
I wouldn't expose this in public API for `MessageId`. It's useful for `Message` but, unless strictly needed, I'd leave it out from `MessasgeId` interface

Reply:
Thanks, Seems need it. in acknowledgeAsync, we use getTopicName(), then use the topicName to find which sub-consumer to do the ack. Also, in UnAckedMessageTracker to remove MessageIds that belongs to a sub-consumer.
will move this into private.

Reply:
Sure, I understand that we need it internally, but we should be careful to expose in API. This method could only be exposed in `MessageIdImpl`